lib/filter/QubitLimitIp.class.php

```diff
@@ -71,28 +72,29 @@ protected function isAllowed()
         foreach ($this->limit as $item) {
             // Ranges are supported, using a comma or a dash
             $limit = preg_split('/[,-]/', $item);
+            $limitBinary = inet_pton(trim($limit[0]));
 
             // Single IP
-            if (1 == count($limit) && $address == $limit[0]) {
+            if (1 == count($limit) && $addressBinary == $limitBinary && strlen($addressBinary) == strlen($limitBinary)) {
                 return true;
             }
 
             // Range
             if (2 == count($limit)) {
                 $limit[0] = trim($limit[0]);
                 $limit[1] = trim($limit[1]);
-
-                $addressLong = ip2long($address);
+                $firstInRangeBinary = inet_pton($limit[0]);
+                $lastInRangeBinary = inet_pton($limit[1]);
 
                 if (
-                    ip2long($limit[0]) <= $addressLong
-                    && ip2long($limit[1]) >= $addressLong
+                    (strlen($addressBinary) == strlen($firstInRangeBinary))
+                     && ($addressBinary >= $firstInRangeBinary && $addressBinary <= $lastInRangeBinary)
                 ) {
                     return true;
                 }
             }
         }
```

Comment:
Syntax checks seem to be failing due to missing new line before return statement. Please add an empty new line before the return here.

Reply:
After adding the extra blank line, CS fixer complained about it:
```
root@ubuntu-focal:/usr/share/nginx/atom# php composer.phar php-cs-fixer fix /usr/share/nginx/atom/lib/filter/QubitLimitIp.class.php -v
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Continue as root/super user [yes]?
> php-cs-fixer: php-cs-fixer 'fix' '/usr/share/nginx/atom/lib/filter/QubitLimitIp.class.php'
PHP CS Fixer 3.59.3 7th Gear by Fabien Potencier, Dariusz Ruminski and contributors.
PHP runtime: 7.4.3-4ubuntu2.23
Running analysis on 1 core sequentially.
You can enable parallel runner and speed up the analysis! Please see usage docs for more information.
Loaded config default from "/usr/share/nginx/atom/.php-cs-fixer.dist.php".
Using cache file ".php-cs-fixer.cache".
Paths from configuration file have been overridden by paths provided as command arguments.
 1/1 [...] 100%

   1) lib/filter/QubitLimitIp.class.php (no_extra_blank_lines)

Fixed 1 of 1 files in 0.125 seconds, 18.00 MB memory used
root@ubuntu-focal:/usr/share/nginx/atom# git diff lib/filter/QubitLimitIp.class.php
diff --git a/lib/filter/QubitLimitIp.class.php b/lib/filter/QubitLimitIp.class.php
index 592277d..cb0dbf6 100644
--- a/lib/filter/QubitLimitIp.class.php
+++ b/lib/filter/QubitLimitIp.class.php
@@ -90,7 +90,6 @@ class QubitLimitIpFilter extends sfFilter
                     (strlen($addressBinary) == strlen($firstInRangeBinary))
                      && ($addressBinary >= $firstInRangeBinary && $addressBinary <= $lastInRangeBinary)
                 ) {
-
                     return true;
                 }
             }
```
...so I went with the CS fixer fix to remove it, and now the syntax check passes 🚀